src/Refine/Flag_Check.cpp

```diff
@@ -338,3 +353,60 @@ bool Check_Curl( const int i, const int j, const int k,
    return Flag;
 
 } // FUNCTION : Check_Curl
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Check_Angular
+// Description :  Check if dh/R at cell (i,j,k) is larger than the minimum angular resolution and smaller
+//                than the maximum angular resolution. Also, check if the cell is in Refined_Rad.
+//
+// Note        :  1. Enabled by the runtime option "OPT__FLAG_ANGULAR"
+//
+// Parameter   :  i,j,k       : Indices of the target element in the patch ptr[0][lv][PID]
+//                lv          : Refinement level of the target patch
+//                PID         : ID of the target patch
+//                CenX        : x coordinate of center of calculating angular resolution
+//                CenY        : y coordinate of center of calculating angular resolution
+//                CenZ        : z coordinate of center of calculating angular resolution
+//                AngRes_Max  : the maximum allowed angular resolution
+//                AngRes_Min  : the minimum allowed angular resolution
+//                Refine_Rad  : the radius of must refined at level (lv)
+//
+// Return      :  0 : if the cell is not in the refine region.
+//                1 : if the minimum angular resolution is     reached or  r <  Refine_Rad
+//                2 : if the minimum angular resolution is not reached and r >= Refine_Rad
+//-------------------------------------------------------------------------------------------------------
+int Check_Angular( const int i, const int j, const int k, const int lv, const int PID,
+                   const real CenX, const real CenY, const real CenZ,
+                   const real AngRes_Max, const real AngRes_Min, const real Refine_Rad )
```

Comment:
The datatype of `CenX/Y/Z`, `AngRes_Max/Min`, and `Refine_Rad` should be `double`